modules/system/console/CreateMigration.php

```diff
@@ -192,13 +193,104 @@ protected function prepareVars(): array
             'version' => $version,
         ];
 
+        if (!empty($model)) {
+            $vars['model'] = $model;
+        }
         if (!empty($table)) {
             $vars['table'] = $table;
         }
 
         return $vars;
     }
 
+    /**
+     * Create vars for model fields mappings so they can be used in update/create stubs
+     */
+    protected function processVars(array $vars): array
+    {
+        $vars = parent::processVars($vars);
+
+        // --model option needed below
+        if (empty($vars['model'])) {
+            return $vars;
+        }
+
+        $vars['fields'] = [];
+
+        try {
+            $fields_path = '$/' . $vars['plugin_url'] . '/models/' . $vars['lower_model'] . '/fields.yaml';
+            $fields = Yaml::parseFile(File::symbolizePath($fields_path));
+        } catch (\Exception $e) {
+            die($e->getMessage());
+        }
+
+        $modelName = $vars['plugin_namespace'] . '\\Models\\' . $vars['model'];
+
+        $vars['model'] = $model = new $modelName();
+
+        foreach (['fields', 'tabs', 'secondaryTabs'] as $type) {
+            if (!isset($fields[$type])) {
+                continue;
+            }
+            if ($type === 'fields') {
+                $fieldList = $fields[$type];
+            } else {
+                $fieldList = $fields[$type]['fields'];
+            }
+
+            foreach ($fieldList as $field => $config) {
+                if (str_contains($field, '@')) {
+                    list($field, $context) = explode('@', $field);
+                }
+
+                $type = $config['type'] ?? 'text';
+
+                if (str_starts_with($field, '_')
+                    or $field === $model->getKeyName()
+                    or str_contains($field, '[')
+                    or in_array($type, ['fileupload','relation','relationmanager','repeater','section','hint'])
```

Comment:
```suggestion
                    or in_array($type, ['fileupload', 'relation', 'relationmanager', 'repeater', 'section', 'hint'])
```
Could we add a style rule for this as well?